pkg/performanceprofile/controller/performanceprofile/hypershift/hypershift.go

```diff
@@ -119,6 +195,16 @@ func (ci *ControlPlaneClientImpl) createInConfigMap(ctx context.Context, obj cli
 	return ci.Client.Create(ctx, cm, opts...)
 }
 
+func (ci *ControlPlaneClientImpl) updateConfigMap(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+	cm := &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: obj.GetName(), Namespace: HostedClustersNamespaceName}}
+	b, err := EncodeManifest(obj, scheme.Scheme)
+	if err != nil {
+		return err
+	}
+	cm.Data = map[string]string{TuningKey: string(b)}
```

Comment:
There's a bug here, we should check what kind of object we're creating.
MC and KC are supposed to be created with the `ConfigKey`

Reply:
I was thinking that we should use Update only for updating performance profile object. I was assuming that non of the other encapsulated objects would need to be updated as part of the E2E tests. But maybe my assumption was wrong

Reply:
We cannot assumed that. please change for `Create` as well since I did the same mistake there.

Reply:
Fixed here:  https://github.com/openshift/cluster-node-tuning-operator/pull/1107